handlers/graph_test.go

```diff
@@ -0,0 +1,346 @@
+package handlers
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/mock"
+	"github.com/swift-sunshine/swscore/config"
+	"github.com/swift-sunshine/swscore/prometheus"
+)
+
+// Setup mock
+
+type promAPIMock struct {
+	mock.Mock
+}
+
+func (o *promAPIMock) Query(ctx context.Context, query string, ts time.Time) (model.Value, error) {
+	args := o.Called(ctx, query, ts)
+	return args.Get(0).(model.Value), args.Error(1)
+}
+
+func (o *promAPIMock) QueryRange(ctx context.Context, query string, r v1.Range) (model.Value, error) {
+	args := o.Called(ctx, query, r)
+	return args.Get(0).(model.Value), args.Error(1)
+}
+
+func (o *promAPIMock) LabelValues(ctx context.Context, label string) (model.LabelValues, error) {
+	args := o.Called(ctx, label)
+	return args.Get(0).(model.LabelValues), args.Error(1)
+}
+
+func (o *promAPIMock) Series(ctx context.Context, matches []string, startTime time.Time, endTime time.Time) ([]model.LabelSet, error) {
+	args := o.Called(ctx, matches, startTime, endTime)
+	return args.Get(0).([]model.LabelSet), args.Error(1)
+}
+
+func setupMocked() (*prometheus.Client, *promAPIMock, error) {
+	config.Set(config.NewConfig())
+	api := new(promAPIMock)
+	client, err := prometheus.NewClient()
+	if err != nil {
+		return nil, nil, err
+	}
+	client.Inject(api)
+	return client, api, nil
+}
+
+func mockQuery(api *promAPIMock, query string, ret *model.Vector) {
+	api.On(
+		"Query",
+		mock.AnythingOfType("*context.emptyCtx"),
+		query,
+		mock.AnythingOfType("time.Time")).
+		Return(*ret, nil)
+	api.On(
+		"Query",
+		mock.AnythingOfType("*context.cancelCtx"),
+		query,
+		mock.AnythingOfType("time.Time")).
+		Return(*ret, nil)
```

Comment:
confusing the way this is formatted - thought Return was just another argument till I realized the line above ends with . and not ,

Reply:
i'll pull that up in the next PR